crypto/x509/pcy_tree.c

```diff
@@ -567,6 +567,10 @@ static int tree_calculate_user_set(X509_POLICY_TREE *tree,
                 | POLICY_DATA_FLAG_EXTRA_NODE;
             node = ossl_policy_level_add_node(NULL, extra, anyPolicy->parent,
                                               tree, 1);
+            if (node == NULL) {
+                ossl_policy_data_free(extra);
+                return 0;
+            }
         }
         if (!tree->user_policies) {
             tree->user_policies = sk_X509_POLICY_NODE_new_null();
```

Comment:
and, isn't the `sk_X509_POLICY_NODE_push` another point of failure?

Reply:
That, and the error handling for the allocation failure of `tree->user_policies = sk_X509_POLICY_NODE_new_null()` seems wrong, because the function is documented to return 1 on success, but returns 1 if the allocation fails.

Reply:
yeah what as well of course!